From c31226a2215b86204d6d1732db291dba91ac16c2 Mon Sep 17 00:00:00 2001
From: Tim Blechmann <tim.blechmann@qt.io>
Date: Thu, 9 Oct 2025 09:47:21 +0800
Subject: [PATCH 336/553] QWidget: initPainter - remove unnecessary const_cast

const_cast tend to be evil and is not necessary when calling a function
that takes a const pointer.

Change-Id: I0e10a5717c636e474a21cb710f1f9d213adeb6b7
Reviewed-by: Eskil Abrahamsen Blomfeldt <eskil.abrahamsen-blomfeldt@qt.io>
(cherry picked from commit ad19d87b7c27bf3f355f3806704367d45b817f28)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/widgets/kernel/qwidget.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/widgets/kernel/qwidget.cpp b/src/widgets/kernel/qwidget.cpp
index 42e6574bd95..d460a5f2e78 100644
--- a/src/widgets/kernel/qwidget.cpp
+++ b/src/widgets/kernel/qwidget.cpp
@@ -13116,7 +13116,7 @@ void QWidget::initPainter(QPainter *painter) const
     const QPalette &pal = palette();
     painter->d_func()->state->pen = QPen(pal.brush(foregroundRole()), 1);
     painter->d_func()->state->bgBrush = pal.brush(backgroundRole());
-    QFont f(font(), const_cast<QWidget *>(this));
+    QFont f(font(), this);
     painter->d_func()->state->deviceFont = f;
     painter->d_func()->state->font = f;
 }
-- 
2.51.2

